scripts/winesapos-setup.sh

```diff
@@ -67,9 +67,28 @@ else
 fi
 echo "Turning on the Mac fan service if the hardware is Apple complete."
 
+# Dialog to ask the user what mirror region they want to use
+# Fetch the list of regions from the Arch Linux mirror status JSON API
+readarray -t arch_mirror_regions <<< $(curl -s https://archlinux.org/mirrors/status/json/ | jq -r '.urls[].country' | sort | uniq | sed '1d')
+chosen_region=$(kdialog --title "winesapOS First-Time Setup" \
+                        --combobox "Select your desired mirror region, \nor press Cancel to use the Arch worldwide mirror:" \
+                        "${arch_mirror_regions[@]}")
+
+# Append the region to /etc/xdg/reflector/reflector.conf (--country C1,C2,C3) if a region was chosen
+if [[ -n "${chosen_region}" ] && [ "${chosen_region}" != "" ]]; then
```

Comment:
I get a syntax error in both Bash and Zsh when testing out this `if` function. This script uses Zsh so here is the relevant error from the `/etc/winesapos/setup_<ISO_DATE>.log` log file.

```
+ chosen_region=Mexico
/home/winesap/.winesapos/winesapos-setup.sh: line 78: syntax error in conditional expression
/home/winesap/.winesapos/winesapos-setup.sh: line 78: syntax error near ']'
/home/winesap/.winesapos/winesapos-setup.sh: line 78: 'if [[ -n "${chosen_region}" ] && [ "${chosen_region}" != "" ]]; then'
```

The setup script then exits immediately after the error.

Reply:
Oops... That was when I was trying to make `kdialog` correct (it causes a blank line which I later just trim with `sed`)